### ISSUE TITLE
CI warnings

### ISSUE THREAD

**Issue by ptheywood** [FLAMEGPU/FLAMEGPU2-docs#18]
Building on CI currently does not error on warnigns. Some of these warnigns should be errors ideally, due to the auto-generated nature of parts of the docs this is impractical to fix / a low priority. 

At some point, it would be good to make the build CI fail on (at least some of) the warnings being output, and resolve as many as possible.

**Comment by ptheywood** [comment]
Down to a single warning when building without the API documentation: 

```
FLAMEGPU2-docs/src/index.rst:9: WARNING: toctree contains reference to nonexisting document 'api/library_root'
```

This can be resolved via CMake generation of the main index.rst? although that will continue to break the out-of-tree builds by modifying files outside of the build directory.

**Comment by ptheywood** [comment]
Some additional warnings have been introduced:

```
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/agent-functions/agent-birth-death.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/agent-functions/agent-communication.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/agent-functions/agent-communication.rst:8: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/agent-functions/defining-agent-functions.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/agent-functions/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/agent-functions/interacting-with-environment.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/agent-functions/random-numbers.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/creating-a-model/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/defining-agents/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/defining-agents/index.rst:68: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/defining-execution-order/dependency-graph.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/defining-execution-order/exit-conditions.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/defining-execution-order/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/defining-execution-order/layers.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/defining-execution-order/submodels.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/defining-messages-communication/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/environment/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/environment/index.rst:76: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/host-functions/agent-operations.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/host-functions/agent-operations.rst:29: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/host-functions/agent-operations.rst:161: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/host-functions/agent-operations.rst:204: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/host-functions/defining-host-functions.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/host-functions/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/host-functions/interacting-with-environment.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/host-functions/interacting-with-environment.rst:64: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/host-functions/random-numbers.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/host-functions/random-numbers.rst:44: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/running-a-simulation/collecting-data.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/running-a-simulation/collecting-data.rst:13: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/running-a-simulation/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/running-a-simulation/initial-state.rst:13: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/running-a-simulation/initial-state.rst:157: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/seatbelts/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/guide/visualisation/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/index.rst:9: WARNING: toctree contains reference to nonexisting document 'api/library_root'
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/quickstart/index.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/quickstart/index.rst:11: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/ptheywood/code/flamegpu/FLAMEGPU2-docs/src/quickstart/index.rst:74: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```

Breathe / Exhale also emits many, many warnings about the autogenereated pages. I'm not sure how trivial these will be to suppress (if possible) to enable Werror builds. 

We could also look at adding GHA annotations as recently added to `FLAMEGPU/FLAMEGPU2`, using probleme matchers to match warnings via regular expressions. the Regex could be constructed to not match certain warnings, but that would add unnecesary complexity (I.e. atleast a few non capturing groups)

**Comment by ptheywood** [comment]
The above warnings are (already) being addressed in #82, though the points still stand about CI Werror issues and potentially adding annotations.
